pkg/services/provisioning/datasources/config_reader.go

```diff
@@ -95,13 +97,8 @@ func (cr *configReader) validateDefaultUniqueness(datasources []*configs) error
 				ds.OrgID = 1
 			}
 
-			if ds.Access == "" {
-				ds.Access = models.DS_ACCESS_PROXY
-			}
-
-			if ds.Access != models.DS_ACCESS_DIRECT && ds.Access != models.DS_ACCESS_PROXY {
-				cr.log.Warn("invalid access value, will use 'proxy' instead", "value", ds.Access)
-				ds.Access = models.DS_ACCESS_PROXY
+			if err := cr.validateAccessAndOrgID(ds); err != nil {
+				return fmt.Errorf("failed to provision %s datasource: %w", ds.Name, err)
```

Comment:
I suggest quoting data source names, in case they contain whitespace.
```suggestion
				return fmt.Errorf("failed to provision %q data source: %w", ds.Name, err)
```